### PR TITLE
fix: allow for linux builds with gcc/clang

### PR DESCRIPTION
**Pull request by dernett**
This fixes the build with Linux. The problems were:
1. Compiler errors when using const variables as array sizes.
2. Not explicitly linking with the C math library.

Tested with gcc/clang.

### CODE REVIEW

File: flappybird.c [hunk 1]
```diff
@@ -141,7 +141,7 @@ typedef struct pipe {
     float gap_y;
 } pipe;
 
-const int Max_Pipes = 10;
+#define   Max_Pipes 10
```
Comment (MarcinKonowalczyk) [inline flappybird.c:144]: seems unrelated to the PR, no?

**Comment by xcom169** [comment]
How can you implement this pull request into your local git clone?

**Comment by lczyk** [comment]
You can directly checkout dernett's branch, or see [this](https://dilankam.medium.com/creating-a-patch-from-github-pull-request-a0381fb3606d) article. In short:
```
curl -L https://github.com/alxyng/flappybird/pull/3.patch | git apply
```

**Comment by xcom169** [comment]
Thanks a lot

**Comment by alxyng** [comment]
Thanks @dernett
